README.md

```diff
@@ -1,10 +1,26 @@
 # streams
 Utilities for working with [Java 8 streams][Stream]. [MoreStreams](#morestreams) provides a few convenient static methods; [KeyedStream](#keyedstream) makes
-working with streams of Map entries readable.
+working with streams of Map entries readable. [ImmutableCollectors](#immutablecollectors) provides collectors for Guava's immutable collections.
```

Comment:
a thought: Maybe make less clear that it's for specifically Guava's immutable collections? It'd be nice to change the implementation of the underlying Map whilst preserving this interface.

To expand a little, one of the things I'm wondering about is whether this collector should use an immutable map, or an unmodifiable map; ImmutableFoo builders are safe to be reused, which means that on construction the whole collection must be copied; it might well prove better to just collect to mutable maps with the same characteristics (LinkedHashMap, LinkedHashSet, ArrayList) and then just wrap with UnmodifiableMap, Set, List.

Don't feel strongly enough about this (and have numbers) to justify this as the initial implementation, but it'd be nice to be able to revisit this assumption later.